service/vehicle.go

```diff
@@ -0,0 +1,84 @@
+package service
+
+import (
+	"encoding/csv"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+
+	"github.com/edgarduran/Go-Dispatch-Bootcamp/models"
+)
+
+func GetAllVehicles() ([]models.Vehicle, error) {
+	f, err := os.Open("vehicles.csv")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer f.Close()
+
+	csvReader := csv.NewReader(f)
+	data, err := csvReader.ReadAll()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Printf("var1 = %T\n", data)
+	var vehicleList []models.Vehicle
+	for _, vehicle := range data {
+		var rec models.Vehicle
+		for j, field := range vehicle {
+			if j == 0 {
+				id, _ := strconv.ParseInt(field, 10, 64)
+				rec.Id = id
+			} else if j == 1 {
+				rec.Make = field
+			} else if j == 2 {
+				rec.Model = field
+			}
+		}
+
+		vehicleList = append(vehicleList, rec)
+	}
+
+	return vehicleList, nil
+}
+
+func GetVehicleById(id int) ([]models.Vehicle, error) {
+	f, err := os.Open("vehicles.csv")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer f.Close()
+
+	csvReader := csv.NewReader(f)
+	data, err := csvReader.ReadAll()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var selectedVehicle []models.Vehicle
+	for _, vehicle := range data {
+		vehicleIntId, _ := strconv.ParseInt(vehicle[0], 10, 64)
+		var i64Id = int64(id)
```

Comment:
This will be overkill, let's imagine you have one million cars, and you are looking for the last one.

with the current logic you have, you will be ending converting 999999 IDs, just for the next iteration being the correct one.

With the approach, I suggested of converting the file data into an actual array of vehicles you won't need to make these extra conversions.

And remember, if you are going to go by this approach, you should never ignore errors.